pulser/waveforms.py

```diff
@@ -207,7 +207,7 @@ def last_value(self) -> float:
         return self._waveforms[-1].last_value
 
     @property
-    def waveforms(self) -> list:
+    def waveforms(self) -> list[Waveform]:
```

Comment:
Please use `List[Waveform]` instead (you'll have to import `List` from `typing`), as this syntax is only available since Python 3.9.